tests/source/enum.rs

```diff
@@ -125,3 +125,16 @@ enum Loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo
 enum Looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong {}
 enum Loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong {}
 enum Loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong { Foo }
+
+// #1046
+pub enum Entry<'a, K: 'a, V: 'a> {
+    // This attribute should stay on the same line.
+    Vacant(
+        #[ stable( feature = "rust1", since = "1.0.0" ) ]   VacantEntry<'a, K, V>,
+    ),
+    // This attribute should be kept on the previous line.
```

Comment:
Sorry, I thought I replied to this comment, but either my comment was lost or I thought I replied when I didn't.

I have generally been very wary of depending on the source code for even very minor input to Rustfmt. It has caused a bunch of bugs in the past, in particular sometimes surprising idempotency bugs. However, we do preserve whitespace around comments (and in some other places) so perhaps there is some scope for compromise here?

As a strawman rule, I suggest we could put attributes on the same line as fields if the whole struct/variant is on one line, and put them on their own line if we would otherwise be multi-line. How does that sound?